### PR TITLE
Use std::vector<T> for some int* and double*

### PR DESCRIPTION
**Pull request by joakim-hove**
Replace many of
```
double * data = util_calloc( ... );
```
with `std::vector<double>`. Just a drive by improvement while working on https://github.com/equinor/ert/pull/2412

### CODE REVIEW

File: libres/lib/res_util/matrix_lapack.cpp [hunk 1]
```diff
@@ -422,36 +380,24 @@ void matrix_dorgqr(matrix_type *A, double *tau,
...
-            /*
-         OK - we could not get the optimal worksize,
-         try again with the minimum.
-      */
-            worksize = n;
-            work = (double *)util_realloc(work, sizeof *work * worksize);
-        } else
-            work = tmp; /* The request for optimal worksize succeeded */
-    }
+    worksize = static_cast<int>(work[0]);
```
Comment (dotfloat) [inline libres/lib/res_util/matrix_lapack.cpp:393]: Unnecessary cast.

File: libres/lib/res_util/matrix_lapack.cpp [hunk 2]
```diff
@@ -383,34 +351,24 @@ void matrix_dgeqrf(matrix_type *A, double *tau) {
...
```
Comment (dotfloat) [inline libres/lib/res_util/matrix_lapack.cpp:364]: Unnecessary cast

File: libres/lib/res_util/matrix_lapack.cpp [hunk 3]
```diff
@@ -347,34 +328,21 @@ int matrix_dsyevx(
...
-                /*
-           OK - we could not get the optimal worksize,
-           try again with the minimum.
-        */
-                worksize = 8 * n;
-                work = (double *)util_realloc(work, sizeof *work * worksize);
-            } else
-                work = tmp; /* The request for optimal worksize succeeded */
-        }
+        worksize = static_cast<int>(work[0]);
```
Comment (dotfloat) [inline libres/lib/res_util/matrix_lapack.cpp:337]: Unnecessary cast

**Review by dotfloat — COMMENTED** [review]
Looks great! Just had some minor nitpicky comments.

File: libres/lib/res_util/matrix_lapack.cpp (hunk 1 above)
Comment (joakim-hove) [reply libres/lib/res_util/matrix_lapack.cpp:393]: This is a `double`&rightarrow; `int` conversion - I thought it was nice to make it explicit; but I am happy to remove the cast if you prefer.

**Comment by joakim-hove** [comment]
@dotfloat : As indicated I thought it made sense to make the `double` &rightarrow; `int` conversion explicit; I think
```C++
double y = ....
int x = y; 
```
will give a `possible loss of precision` warning. But I have no strong feelings and will do as requested.

**Comment by dotfloat** [comment]
Yeah, okay. In that case, let's keep the casting

**Review by dotfloat — APPROVED** [review]
:+1:

**Comment by joakim-hove** [comment]
test ert please

**Comment by joakim-hove** [comment]
Why are several tests pending?

**Comment by oyvindeide** [comment]
> Why are several tests pending?

Can happen if there are conflicts, but does not appear to be the case here

**Comment by dafeda** [comment]
Could you try rebasing on main to see if that fixes the forever-pending tests?
I asked in the #ert-dev Slack-channel and the pending might be caused by changes to some required tests.

**Comment by joakim-hove** [comment]
> Could you try rebasing on main to see if that fixes the forever-pending tests?

Done

> I asked in the #ert-dev Slack-channel and the pending might be caused by changes to some required tests.

Thank you.

**Comment by dafeda** [comment]
This can be merged no, no need to re-run the `ERT - pytest`.

**Pull request merged**